cs/ccxt/base/Throttler.cs

```diff
@@ -19,25 +21,36 @@ public Throttler(dict config)
             {"delay", 0.001},
             {"cost", 1.0},
             {"tokens", 0},
-            {"maxCapacity", 2000},
+            {"maxLimiterRequests", 2000},
             {"capacity", 1.0},
+            {"algorithm", "leakyBucket"},
+            {"rateLimit", 0.0},
+            {"windowSize", 60000.0},
+            // maxWeight should be set in config parameter for rolling window algorithm
         };
         this.config = extend(this.config, config);
 
     }
 
-    private async Task loop()
+    private async Task leakyBucketLoop()
     {
         var lastTimestamp = milliseconds();
         while (this.running)
         {
             // do we need this check here?
-            if (this.queue.Count == 0)
+            lock (queueLock)
```

Comment:
@carlosmiei I remember you had further experience with locks and some further know-how when you worked on slim concurrent locks.
wdyt about this? is it ok to be a simple lock here